ldclient.go

```diff
@@ -134,6 +134,8 @@ func MakeCustomClient(sdkKey string, config Config, waitFor time.Duration) (*LDC
 				config.Logger.Println("Timeout exceeded when initializing LauncDarkly client.")
 				return &client, ErrInitializationTimeout
 			}
+
+			go func() { <-ch }() // Don't block the updateProcessor when not waiting
```

Comment:
Should we not just close that channel when a timeout happens?

Reply:
The problem with closing the channel is that when the client is eventually initialized, it will try to signal this on the channel. If it is closed, then it will panic with `send on closed channel`.